rest-server/src/main/java/org/brooklyncentral/catalog/validations/CatalogValidator.java

```
@@ -0,0 +1,74 @@
+ * under the License.
+ */
+
+import com.google.common.base.Optional;
+import org.brooklyncentral.catalog.scrape.CatalogScraper;
+import org.brooklyncentral.catalog.scrape.CatalogScraperHelper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+
+public class CatalogValidator {
+
+
+    private static final Logger LOG = LoggerFactory.getLogger(CatalogScraper.class);
+
+    public static boolean validate(String repoUrl) {
+        LOG.info("Start validation...");
+
+        List<String> catalogItemRepoUrls = CatalogScraper.parseDirectoryYaml(repoUrl);
+
+        boolean successfullyValidated = true;
+        for (String catalogItemRepoUrl : catalogItemRepoUrls) {
+            successfullyValidated = successfullyValidated && validateCatalogItem(catalogItemRepoUrl);
+        }
+
+        LOG.info("Validation complete");
+
+        return successfullyValidated;
+    }
+
+    public static boolean validateCatalogItem(String repoUrl) {
+        String[] urlTokens = repoUrl.split("/");
+        String repoName = urlTokens[urlTokens.length - 1];
+        String author = urlTokens[urlTokens.length - 2];
+
+        LOG.info("Validating catalog " + author + "/" + repoName);
+
+        Optional<String> description = Optional.absent();
+        Optional<String> catalogBomString = Optional.absent();
+        boolean isValid = true;
+        try {
+            description = CatalogScraperHelper.getGithubRawText(repoUrl, "README.md", true);
+            catalogBomString = CatalogScraperHelper.getGithubRawText(repoUrl, "catalog.bom", true);
+            LOG.info("Successfull validation");
+        } catch (IllegalStateException e) {
+            if (!description.isPresent()) {
+                LOG.info("Validation failed - README.md is required");
+            }
+            if (!catalogBomString.isPresent()) {
+                LOG.info("Validation failed - catalog.bom is required");
+            }
+            isValid = false;
```

Why not propagate the exception? It would make the check easier on [line 41](https://github.com/brooklyncentral/catalog-server/pull/21/files#diff-1752031ff1a83e831991bf8d57338815R41)